### PR TITLE
[NATIVECPU] Fix build errors when -Werror=on

### PR DESCRIPTION
**Pull request by wenju-he**

### CODE REVIEW

**Comment by wenju-he** [comment]
@uwedolinsky @PietroGhg could you please review?

File: source/adapters/native_cpu/kernel.cpp [hunk 1]
```diff
@@ -171,6 +172,10 @@ urKernelGetSubGroupInfo(ur_kernel_handle_t hKernel, ur_device_handle_t hDevice,
     // todo: set proper values
     return ReturnValue(0);
   }
+  case UR_KERNEL_SUB_GROUP_INFO_FORCE_UINT32: {
```
Comment (pbalcer) [inline source/adapters/native_cpu/kernel.cpp:175]: this is not really a real value.
`default: {
} break;`
?

Comment (kbenzie) [reply source/adapters/native_cpu/kernel.cpp:175]: This should be an unreachable case so I agree with the second suggestion.

Comment (wenju-he) [reply source/adapters/native_cpu/kernel.cpp:175]: done

Comment (pbalcer) [reply source/adapters/native_cpu/kernel.cpp:175]: could be useful to have something like unreachable in common, but otherwise lgtm.

Comment (kbenzie) [reply source/adapters/native_cpu/kernel.cpp:175]: Would `std::abort()` suffice?

**Review by kbenzie — APPROVED** [review]

**Pull request merged**